app/src/main/java/com/lucas/medicaltools/MedicalToolAdapter.kt

```diff
@@ -0,0 +1,93 @@
+package com.lucas.medicaltools
+
+import android.content.Context
+import android.graphics.Color
+import android.net.Uri
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.*
+import androidx.core.content.contentValuesOf
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.snackbar.Snackbar
+import com.lucas.medical_equip.repository.MedicalTool
+import com.lucas.medicaltools.databinding.MedicalToolItemBinding
+import com.squareup.picasso.Picasso
+import java.util.*
+import kotlin.collections.ArrayList
+
+class MedicalToolAdapter(private var medicalTools: List<MedicalTool>) : RecyclerView.Adapter<MedicalToolAdapter.ViewHolder>(), Filterable {
```

Comment:
Delete Filterable as you'll make use of MVI to push your filter text to your presenter layer, and that will filter the information received.